gcloud/logging/logging_service_v2_api.py

```diff
@@ -38,47 +38,101 @@
+                 service_path=SERVICE_ADDRESS,
+                 port=DEFAULT_SERVICE_PORT,
+                 channel=None,
+                 ssl_creds=None,
+                 scopes=_ALL_SCOPES,
+                 retrying_override=None,
+                 bundling_override=None,
+                 timeout=30,
+                 app_name=None,
+                 app_version=None):
+        """Constructor.
+
+        Args:
+          :keyword service_path: The DNS of the API remote host.
+          :type service_path: string
+          :keyword port: The port on which to connect to the remote host.
+          :type port: int
+          :keyword channel: A Channel object through which to make calls.
+          :type channel: A grpc.beta.implementations.Channel object
+          :keyword ssl_creds: A ClientCredentials for use with an SSL-
+            enabled channel
+          :type ssl_creds: A grpc.beta.implementations.ClientCredentials
+            object
+          :keyword retrying_override: A dictionary that overrides default
+            retrying settings. ``retrying_override`` maps method names
+            (e.g., 'list_foo') to custom RetryOptions objects, or to None.
+            A value of None indicates that the method in question should not
+            retry.
+          :type retrying_override: dict
+          :keyword bundling_override: A dictionary that overrides default
+            bundling settings. ``bundling_override`` maps bundling method
+            names (e.g., 'publish_foo') to custom BundleOptions objects, or to
+            None. It is invalid to have a key for a method that is not
+            bundling-enabled. A value of None indicates that the method in
+            question should not bundle.
+          :type bundling_override: dict
+          :keyword timeout: The default timeout, in seconds, for calls made
+            through this client
+          :type timeout: int
+          :keyword app_name: The codename of the calling service.
+          :type app_name: string
+          :keyword app_version: The version of the calling service.
+          :type app_version: string
+        """
+        if app_name is None:
+            app_name = 'gax'
+        if app_version is None:
+            app_version = google.gax.__version__
+        bundling_override = bundling_override or dict()
+        retrying_override = retrying_override or dict()
+        config_filename = os.path.join(
+            os.path.dirname(__file__), 'logging_service_v2_api.yaml')
+        with open(config_filename, 'r') as api_yaml:
+            self._defaults = api_callable.construct_settings(
+                yaml.load(api_yaml.read()), bundling_override,
+                retrying_override, config.STATUS_CODE_NAMES, timeout)
+        google_apis_agent = '{}-{}/{}/gax-{}/{}'.format(
+            app_name, app_version, self._CODE_GEN_NAME_VERSION,
+            google.gax.__version__,
+            'python-{}'.format(platform.python_version()))
```

Comment:
(Sorry I just noticed this.)

You could just change the original format string to `'{}-{}/{}/gax-{}/python-{}' and then only have to format once.